hal/src/thumbv7em/pukcc.rs

```diff
@@ -21,8 +21,10 @@
 pub mod c_abi;
 pub mod curves;
 
+use core::iter::{once, repeat};
+
 use crate::pac::MCLK;
-use c_abi::{u4, CryptoRamSlice, Service};
+use c_abi::{u2, u4, CryptoRamSlice, Service};
```

Comment:
Why did you alias `u32` to `u4` and `u16` to `u2`? I think this is quite confusing. In Rust I would expect any `ux` type to have `x` bits and not `x` bytes.

Reply:
These aliases are used by PUKCC C ABI in reverse engineered C HAL, so I sticked to it. I considered this a good thing to avoid mistakes while porting stuff, the same goes for casing in names of fields in `c_abi.rs`. This might be useful for someone inspecting this code for differences against the C reference HAL in cases something misbehaves.
`CryptoLib_typedef_pb.h`:
```c
typedef unsigned char u1;
typedef signed char   s1;
typedef u1 *          pu1;   // pointer near data unsigned
typedef s1 *          ps1;   // pointer near data unsigned
typedef u1 *          pfu1;  // pointer far data unsigned
typedef s1 *          pfs1;  // pointer far data signed
typedef u1 *          pcu1;  // pointer near code unsigned
typedef s1 *          pcs1;  // pointer near code signed
typedef u1 *          pfcu1; // pointer far code unsigned
typedef s1 *          pfcs1; // pointer far code signed

typedef unsigned short u2;
typedef signed short   s2;
typedef u2 *           pu2;   // pointer near data unsigned
typedef s2 *           ps2;   // pointer near data unsigned
typedef u2 *           pfu2;  // pointer far data unsigned
typedef s2 *           pfs2;  // pointer far data signed
typedef u2 *           pcu2;  // pointer near code unsigned
typedef s2 *           pcs2;  // pointer near code signed
typedef u2 *           pfcu2; // pointer far code unsigned
typedef s2 *           pfcs2; // pointer far code signed

typedef unsigned long int u4;
typedef signed long int   s4;
typedef u4 *              pu4;   // pointer near data unsigned
typedef s4 *              ps4;   // pointer near data unsigned
typedef u4 *              pfu4;  // pointer far data unsigned
typedef s4 *              pfs4;  // pointer far data signed
typedef u4 *              pcu4;  // pointer near code unsigned
typedef s4 *              pcs4;  // pointer near code signed
typedef u4 *              pfcu4; // pointer far code unsigned
typedef s4 *              pfcs4; // pointer far code signed
```